dislib/model_selection/_search.py

```diff
@@ -0,0 +1,387 @@
+            results['std_%s' % key_name] = array_stds
+
+            if rank:
+                results["rank_%s" % key_name] = np.asarray(
+                    rankdata(-array_means, method='min'), dtype=np.int32)
+
+        _store('master_fit_time', master_fit_time)
+        _store('master_score_time', master_score_time)
+        # Use one MaskedArray and mask all the places where the param is not
+        # applicable for that candidate. Use defaultdict as each candidate may
+        # not contain all the params
+        param_results = defaultdict(partial(MaskedArray,
+                                            np.empty(n_candidates, ),
+                                            mask=True,
+                                            dtype=object))
+        for cand_i, params in enumerate(candidate_params):
+            for name, value in params.items():
+                # An all masked empty array gets created for the key
+                # `"param_%s" % name` at the first occurrence of `name`.
+                # Setting the value at an index also unmasks that index
+                param_results["param_%s" % name][cand_i] = value
+
+        results.update(param_results)
+        # Store a list of param dicts at the key 'params'
+        results['params'] = candidate_params
+
+        for scorer_name in scorers.keys():
+            _store('test_%s' % scorer_name, test_scores[scorer_name],
+                   splits=True, rank=True)
+
+        return results
+
+    def _infer_scorers(self):
+        estimator = self.estimator
+        scoring = self.scoring
+        refit = self.refit
+        if scoring is None or callable(scoring):
+            scorers = {"score": check_scorer(estimator, scoring)}
+            refit_metric = 'score'
+            self.multimetric_ = False
+        elif isinstance(scoring, dict):
+            scorers = {key: check_scorer(estimator, scorer)
+                       for key, scorer in scoring.items()}
+            if refit is not False and (
+                    not isinstance(refit, str) or
+                    refit not in scorers) and not callable(refit):
+                raise ValueError("For multi-metric scoring, the parameter "
+                                 "refit must be set to a scorer key or a "
+                                 "callable to refit an estimator with the "
+                                 "best parameter setting on the whole "
+                                 "data and make the best_* attributes "
+                                 "available for that metric. If this is "
+                                 "not needed, refit should be set to "
+                                 "False explicitly. %r was passed."
+                                 % refit)
+            refit_metric = refit
+            self.multimetric_ = True
+        else:
+            raise ValueError('scoring is not valid')
+
+        return scorers, refit_metric
+
+
+class GridSearchCV(BaseSearchCV):
```

Comment:
Have you checked that the documentation is correctly built?

Reply:
I had not. Now I checked and added the required files.